cachi2/interface/cli.py

```diff
@@ -321,3 +325,17 @@ def _get_build_config(output_dir: Path) -> BuildConfig:
             "Please use a directory populated by a previous fetch-deps command."
         )
     return BuildConfig.parse_raw(build_config_json.read_text())
+
+
+def _is_dirty_git_repository(source_dir: Path) -> bool:
+    try:
+        repo = Repo(source_dir)
+    except InvalidGitRepositoryError:
+        # not a git repository => clean
+        return False
+
+    if repo.is_dirty(untracked_files=True):
+        log.info(repo.git.status("--porcelain"))
```

Comment:
Currently the output looks like this:

```
❯ cachi2 fetch-deps pip
2023-03-30 11:39:24,910 INFO  M tox.ini
?? .markdownlint.yaml
?? tests/unit/data/gomod-mocks/
2023-03-30 11:39:24,910 ERROR UncleanSourceRepo: The repository is considered dirty
Error: UncleanSourceRepo: The repository is considered dirty
  Fetching dependencies is not supported for unclean git repositories.
  The source repository must be clean (no untracked files, no changes).
```

Let's make sure the git status output will be properly aligned and log something to explain what the output is. And we should probably log this at ERROR level, @fepas can confirm :slightly_smiling_face: 

```
2023-03-30 11:39:24,910 ERROR The repository is not clean (git status --porcelain):
 M tox.ini
?? .markdownlint.yaml
?? tests/unit/data/gomod-mocks/
```

Reply:
What do you think about this output ?
(Skip temporary ugly functional programming)

```
log.error(f"Changes to be committed: {list(map(lambda x: x.a_path, list(repo.index.diff('HEAD'))))}")
log.error(f"Changes not staged for commit: {list(map(lambda x: x.a_path, list(repo.index.diff(None))))}")
log.error(f"Untracked files: {repo.untracked_files}")
```

Reply:
IMO one file per line is easier to read, git status is pretty good in this regard. For my part, as long as the git status output is separated by a newline from the log message, it's readable enough